### PR TITLE
EIRH-52: Sorting UI element.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Adds sorting UI widget and sorting indexes

# Review By (Date)
- EOW

# Criticality
- Med/High

# Urgency
- Med/High

# Review Tasks

1. Review code 
2. Load up the UI
3. Change the sorting and review the results (make sure sort is right)
4. Change the sorting and make sure the URL changes
5. Copy and paste the URL into a new window for deep link test

### CODE REVIEW

File: src/eventshub/src/component/CustomHits/CustomHits.tsx
Comment (sherakama) [inline src/eventshub/src/component/CustomHits/CustomHits.tsx:1]: Just changed the export on this to be consistent with the others

File: src/eventshub/src/component/Search/Search.tsx [hunk 1]
```diff
@@ -59,14 +60,6 @@ export const Search = () => {
...
   // ------------------------------------------------------------------------
 
-  // Change the index when the active index changes.
-  useEffect(() => {
-    const swapIndex = async () => {
-      await algoliaClient.initIndex(activeIndex);
-    }
-    swapIndex();
-  }, [activeIndex, dispatch]);
-
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:70]: Not required and moved to the `<Index>` component.

File: src/eventshub/src/component/Search/Search.tsx [hunk 2]
```diff
@@ -92,13 +85,16 @@ export const Search = () => {
 
   // Routing object to handle router URLs.
   const routing = {
-    router: history(),
+    router: history({
+      cleanUrlOnDispose: false,
+    }),
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:91]: Clears up a warning.

File: src/eventshub/src/component/Search/Search.tsx [hunk 3]
```diff
@@ -92,13 +85,16 @@ export const Search = () => {
...
     stateMapping: {
       stateToRoute(uiState:UiState) {
         // Get the first key of uiState
-        const indexUiState = uiState[Object.keys(uiState)[0]];
+        const indexUiState = uiState[Object.keys(uiState)[1]];
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:95]: Index component moved them one deeper

File: src/eventshub/src/component/Search/Search.tsx [hunk 4]
```diff
@@ -109,7 +105,7 @@ export const Search = () => {
           // Active tab.
           tab: tab,
           // Sort order.
-          // sortBy: TBD
+          sortBy: sort === 'relevance' ? undefined : sort,
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:109]: Add sortBy URL param

File: src/eventshub/src/component/Search/Search.tsx [hunk 5]
```diff
@@ -182,50 +178,52 @@ export const Search = () => {
...
-                  />
-                );
-              })}
-            </FlexBox>
-            <FlexBox direction='col' className='er-col-span-9'>
-              <LoadingIndicator />
-              <CustomHits
-                hitComponent={SearchItemComponent}
-                noResultsMessage={`We're sorry, but no results were found matching your search criteria. Please try again with different keywords or filters.`}
+          <Index indexName={activeIndex} indexId={activeIndex}>
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:189]: This is the real only change here in that I wrapped the contents in the `<Index>` component.

File: src/eventshub/src/component/SortBy/SortBy.tsx
Comment (sherakama) [inline src/eventshub/src/component/SortBy/SortBy.tsx:1]: New SortBy UI widget

File: src/eventshub/src/hooks/useAppState.ts [hunk 6]
```diff
@@ -19,17 +20,26 @@ export default function useAppState() {
...
       }
+      // Set the sorting
+      if (sortBy && sortOptions.includes(sortBy as SortOptions)) {
+        dispatch({ type: 'app/setSortBy', payload: sortBy });
+      }
+      // Set the query
+      const q = urlParams.get('q');
+      if (q) {
+        dispatch({ type: 'app/setQuery', payload: q });
+      }
```
Comment (sherakama) [inline src/eventshub/src/hooks/useAppState.ts:41]: Setting the App state before the search UI loads.

**Review by rebeccahongsf — APPROVED** [review]
Works great! Thank you @sherakama. I'll go ahead and roll this in so I can theme 👍

**Pull request merged**